File: Clase_01/Practica_01/Numero.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Practica_01
{
    //EJERCICIO 2 🎉

    /// <summary>
    /// Representa un número. Un número tiene un valor.
    /// </summary>
    public class Numero : Comparable
    {
        // Atributos 
        private int valor;

        //Constructor 
        public Numero(int valor)
        {
            this.valor = valor;
        }

        /// <summary>
        /// Describe el valor del número que recibe el mensaje.
        /// </summary>
        /// <returns>int</returns>
        public int getValor()
        {
            return valor;
        }

        public override string ToString()
        {
            return $"{getValor()}"; // return getValor().ToString();
        }

        // Implementación de los métodos de la interfaz comparable.

        public bool sosIgual(Comparable valor)
        {
            return this.valor == ((Numero)valor).getValor();
        }

        public bool sosMenor(Comparable valor) 
        {
            return this.valor < ((Numero)valor).getValor();
        }

        public bool sosMayor(Comparable valor)
        {
            return this.valor > ((Numero)valor).getValor();
        }
    }
}

File: interfaces/src/main/java/numero/Numero.java
package numero;

import comparable.Comparable;

public class Numero implements Comparable {
    private int valor;

    public Numero(int valor) {
        this.valor = valor;
    }

    public int getValor(){
        return valor;
    }

    @Override
    public Boolean sosIgual(Comparable comparable) {
        return this.getValor() == ((Numero)comparable).getValor();
    }

    @Override
    public Boolean sosMenor(Comparable comparable) {
        return this.getValor() < ((Numero)comparable).getValor();
    }

    @Override
    public Boolean sosMayor(Comparable comparable) {
        return this.getValor() > ((Numero)comparable).getValor();
    }

    @Override
    public String toString() {
        return String.format("%d", getValor());
    }
}
